docs/docsite/rst/porting_guides/porting_guide_core_2.19.rst

```diff
@@ -633,6 +633,35 @@ Valid options are:
 Multiple options can be combined by separating them with commas.
 
 
+Displaying warning when undefined variables in var_files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In previous versions of ``ansible-core``, undefined variables used while specifying file paths in var_files were silently ignored and did not trigger warning.
+This is now changed and a warning will be displayed when undefined variables while specifying file paths in var_files are encountered.
+
+.. code-block:: yaml+jinja
+
+    - hosts: all
+      vars_files:
+        - "{{ inventory_dir }}/vars_files/bar.yml"
+
+.. code-block:: ansible-output
+
+    PLAYBOOK: foo.yml ************************************************************************
+    1 plays in foo.yml
+    [WARNING]: skipping vars_file item due to an undefined variable
+    Origin: /examples/foo.yml:6:7
+
+    4
+    5   vars_files:
+    6     - "{{ inventory_dir }}/vars_files/bar.yml"
+            ^ column 7
+
+In the above example, the warning is displayed because the variable ``inventory_dir`` is undefined at play level.
+This is because ``inventory_dir`` is a special host scoped variable which will be evaulated at task level and not at play level.
+The ``inventory_dir`` will work as expected when it is used in a task level variable, which makes vars defined in vars_files available to the tasks.
```

Comment:
```suggestion
In the preceding example, the warning is displayed because ``inventory_dir`` is a host-scoped variable that is evaluated at the task level, not at the play level where ``vars_files`` is processed.
While ``inventory_dir`` does not work in ``vars_files``, it can be used in task-level variables where the vars from ``vars_files`` are already available.
```